CI/setup-windows-sdk.sh

```diff
@@ -102,22 +102,7 @@ while true; do
     "mingw-w64-${BUILDCOMPONENT}-qt6-tools" \
     "mingw-w64-${BUILDCOMPONENT}-qt6-5compat" \
     "mingw-w64-${BUILDCOMPONENT}-angleproject" \
-    "mingw-w64-${BUILDCOMPONENT}-qtkeychain-qt6"; then
-      break
-  fi
-
-  if [ $pacman_attempts -eq 10 ]; then
-    exit 7
-  fi
-  pacman_attempts=$((pacman_attempts +1))
-
-  echo "=== Some packages failed to install, waiting and trying again ==="
-  sleep 10
-done
-
-pacman_attempts=1
-while true; do
-  if /usr/bin/pacman -Su --needed --noconfirm \
+    "mingw-w64-${BUILDCOMPONENT}-qtkeychain-qt6" \
```

Comment:
I've cleaned this up to do everything in one loop - and improved the error reporting process.